taskcluster/mozillavpn_taskgraph/transforms/beetmover.py

```diff
@@ -13,33 +13,78 @@
 @transforms.add
 def add_beetmover_worker_config(config, tasks):
     for task in tasks:
-        app_name = "vpn"
         worker_type = task["worker-type"]
         is_relpro = (
             config.params["level"] == "3"
             and config.params["tasks_for"] in task["run-on-tasks-for"]
         )
         bucket = "release" if is_relpro else "dep"
-        build_id = config.params["moz_build_date"]
+        moz_build_date = config.params["moz_build_date"]
         build_type = task["attributes"]["build-type"]
         build_os = os.path.dirname(build_type)
-        app_version = config.params["version"]
-        candidates_path = os.path.join(
-            "pub", app_name, "candidates", app_version, build_id, build_os
+        shipping_phase = config.params.get("shipping_phase", "")
+
+        def get_version_from_release_branch_head_ref():
+            return config.params["head_ref"].split("/")[-1]
+
+        app_version = (
+            config.params["version"]
+            if config.params["version"]
+            else get_version_from_release_branch_head_ref()
         )
+
+        def get_candidates_path():
+            if build_type == "addons/opt":
+                return os.path.join(
+                    "pub",
+                    "vpn",
+                    "addons",
+                    "releases" if shipping_phase.startswith("ship") else "candidates",
+                    moz_build_date,
+                )
+            return os.path.join(
+                "pub",
+                "vpn",
+                "candidates",
+                f"{app_version}-candidates",
+                f"build{config.params['build_number']}",
+                build_os,
+            )
+
+        candidates_path = get_candidates_path()
         destination_paths = [candidates_path]
+
+        if shipping_phase == "ship-addons":
+            destination_paths.append(
+                os.path.join(
+                    "pub",
+                    "vpn",
+                    "addons",
+                    "releases",
+                    "latest",
+                )
+            )
+
         archive_url = (
             "https://ftp.mozilla.org/" if is_relpro else "https://ftp.stage.mozaws.net/"
         )
-        task_description = f"This {worker_type} task will upload a {build_os} release candidate for v{app_version} to {archive_url}{candidates_path}/"
+
+        def get_task_description():
+            if build_type == "addons/opt":
+                return f"This {worker_type} task will upload the {task['name']} to {archive_url}{candidates_path}/"
+            if shipping_phase == "ship-client":
+                return f"This {worker_type} task will copy build {config.params['build_number']} from candidates to releases"
+            return f"This {worker_type} task will upload a {build_os} release candidate for v{app_version} to {archive_url}{candidates_path}/"
+
+        task_description = get_task_description()
```

Comment:
Personally I would prefer just using an if statement over these inline functions, but not a strong blocker.

Reply:
That cleaned up nicely, thanks